pkg/upgrade/upgrades/alter_jobs_add_job_type.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package upgrades
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/clusterversion"
+	"github.com/cockroachdb/cockroach/pkg/keys"
+	"github.com/cockroachdb/cockroach/pkg/security/username"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/systemschema"
+	"github.com/cockroachdb/cockroach/pkg/upgrade"
+)
+
+const addTypeColumnStmt = `
+ALTER table system.jobs
+ADD COLUMN IF NOT EXISTS type STRING
```

Comment:
drive-by: what would you think about calling it `job_type` since type is a reserved word in some/many places?

that's what the revamp RFC proposed here: https://github.com/cockroachdb/cockroach/pull/82638/files#diff-d65357c15cffdb009c1b3fd0f6456265d25b94758f326207c531b8f734b59164R126

Reply:
Sgtm 👍🏽